### PR TITLE
OKTA-394749 Implement "Skip" functionality

### PR DESCRIPTION
**Pull request by andriizhegurov-okta**

### CODE REVIEW

**Comment by bryanapellanes-okta** [comment]
There appears to be conflicts that need to be resolved.  Also, should this target the `idx-auth-layer` branch?

File: src/Okta.Idx.Sdk.UnitTests/IdxClientShould.cs [hunk 1]
```diff
@@ -1013,5 +1013,50 @@ public async Task CallInteractOnStartSignInWidget()
             Assert.Equal("this is a test interaction handle", widgetResponse.IdxContext.InteractionHandle);
             Assert.Equal(1, mockHttpMessageHandler.CallCounts["/v1/interact"]);
         }
+
+        [Fact]
+        public async Task ProcessInfoMessageResponse()
```
Comment (laura-rodriguez) [inline src/Okta.Idx.Sdk.UnitTests/IdxClientShould.cs:1018]: I suggest we keep all "parsing" UTs in the `IdxClientLayer1Should` class and use this one for the "layer 3" tests (public IdxClient methods).

File: src/Okta.Idx.Sdk/Extensions/IdxClientExtensions.cs [hunk 2]
```diff
@@ -13,22 +13,28 @@ internal static class IdxClientExtensions
...
 
             return await remediationOption
                     .ProceedAsync(request, cancellationToken);
         }
 
         internal static bool ContainsRemediationOption(this IIdxResponse response, string remediationType) =>
                     response.Remediation.RemediationOptions.Any(x => x.Name == remediationType);
-
+
+        internal static IRemediationOption FindRemediationOption(this IIdxResponse response, string remediationType, bool throwIfNotFound = false)
```
Comment (laura-rodriguez) [inline src/Okta.Idx.Sdk/Extensions/IdxClientExtensions.cs:25]: Should we move this to the IdxResponse extensions?

File: src/Okta.Idx.Sdk/IIdxClient.cs [hunk 3]
```diff
@@ -143,5 +142,13 @@ public interface IIdxClient : IOktaClient
         /// <param name="cancellationToken">The cancellation token.</param>
         /// <returns>The widget sign in response.</returns>
         Task<WidgetSignInResponse> StartWidgetSignInAsync(CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Chooses 'skip' remediation option
```
Comment (laura-rodriguez) [inline src/Okta.Idx.Sdk/IIdxClient.cs:147]:
```suggestion
        /// Skips optional authenticators
```

File: src/Okta.Idx.Sdk/IIdxResponse.cs [hunk 4]
```diff
@@ -52,6 +52,11 @@ public interface IIdxResponse : IResource
         /// <returns>An IdxResponse.</returns>
         Task<IIdxResponse> CancelAsync(CancellationToken cancellationToken = default);
 
+        /// <summary>
+        /// Gets the current authenticator
+        /// </summary>
+        IAuthenticatorEnrollmentData CurrentAuthenticator { get; }
```
Comment (laura-rodriguez) [inline src/Okta.Idx.Sdk/IIdxResponse.cs:58]: What's the diff between this prop and `CurrentAuthenticatorEnrollment`? they both seem to be the same type?

File: src/Okta.Idx.Sdk/IMessage.cs [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+namespace Okta.Idx.Sdk
+{
+    /// <summary>
+    /// An interface to represent a sigle Idx message.
+    /// </summary>
+    public interface IMessage : IResource
+    {
+        string Text { get; }
+        string Class { get; }
```
Comment (laura-rodriguez) [inline src/Okta.Idx.Sdk/IMessage.cs:19]: When does the server return this object? Also, we shouldn't expose UI props so we might remove `Class`.

File: src/Okta.Idx.Sdk/IdxClient.cs [hunk 6]
```diff
@@ -686,6 +686,41 @@ public async Task<AuthenticationResponse> SelectChallengeAuthenticatorAsync(Sele
...
+                StateHandle = introspectResponse.StateHandle,
+            };
+
+            var skipResponse = await skipOption.ProceedAsync(skipRequest, cancellationToken);
+            if (skipResponse.IdxMessages.Messages.Any())
+            {
+                return new AuthenticationResponse
+                {
+                    AuthenticationStatus = AuthenticationStatus.MessageToUser,
+                    MessageToUser = skipResponse.IdxMessages.Messages.First().Text,
```
Comment (laura-rodriguez) [inline src/Okta.Idx.Sdk/IdxClient.cs:706]: I'm curious about this status, when you can end up with `MessageToUser`?

**Review by laura-rodriguez — COMMENTED** [review]
I left a few questions and comments for us to review before approval.

File: src/Okta.Idx.Sdk/IIdxResponse.cs (hunk 4 above)
Comment (andriizhegurov-okta) [reply src/Okta.Idx.Sdk/IIdxResponse.cs:58]: They are of the same type but names are different, one is being returned when one-step-auth is configured and another for two-steps-auth

Comment (andriizhegurov-okta) [reply src/Okta.Idx.Sdk/IIdxResponse.cs:58]: they are used for password recovery flow

File: src/Okta.Idx.Sdk/IdxClient.cs (hunk 6 above)
Comment (laura-rodriguez) [reply src/Okta.Idx.Sdk/IdxClient.cs:706]: Rename Status to `Terminal`

**Review by laura-rodriguez — APPROVED** [review]

**Pull request merged**